arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,226 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class CreateSymbolStylesFromWebStylesViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap(featureLayer: featureLayer)
+            mapView.setViewpoint(
+                AGSViewpoint(
+                    latitude: 34.28301,
+                    longitude: -118.44186,
+                    scale: 1e4
+                )
+            )
+        }
+    }
+    /// The button to show legend table.
+    @IBOutlet var legendBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Instance properties
+    
+    /// The observation on the map view's `mapScale` property.
+    var mapScaleObservation: NSKeyValueObservation?
+    /// The symbols created from the symbol style to render the feature layer.
+    var pointSymbols = [(category: SymbolCategory, symbol: AGSSymbol)]()
```

Comment:
Good point. I forgot to sort the array after the dispatch group calls back.

I compared `Dict` vs `KeyValuePairs` vs `Array<Tuple>` and used the latter with a few reasons

- Since there is no need to look up a key in this sample, a linear array is sufficient
- Named tuple makes it more readable to access each element in a tuple
- Only need to sort once

Taken the other suggestions into consideration, I'll probably consider creating an observable array of symbols, and whenever it gets changed, re-sort the array and let the table observe the change.

Reply:
And it really makes me think of using [`UITableViewDiffableDataSource` ](https://developer.apple.com/documentation/uikit/uitableviewdiffabledatasource) in U11 with iOS 13, to handle these dynamic async tables. Will explore later.

Reply:
For cases like this, where the underlying data is only assigned once, we can use segue actions starting with U11. The would simplify things somewhat since we can make the underlying data immutable.

Reply:
That is nice! 🤩  Will try out soon.